alembic/versions/20230905_2b672c6fb2b9_type_coerce_collection_settings.py

```diff
@@ -0,0 +1,77 @@
+"""Type coerce collection settings
+
+Revision ID: 2b672c6fb2b9
+Revises: 0df58829fc1a
+Create Date: 2023-09-05 06:40:35.739869+00:00
+
+"""
+import json
+from typing import Callable, Dict
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "2b672c6fb2b9"
+down_revision = "0df58829fc1a"
+branch_labels = None
+depends_on = None
+
+
+def _bool(value):
+    return value in ("true", "True", True)
+
+
+# All the settings types that have non-str types
+ALL_SETTING_TYPES: Dict[str, Callable] = {
+    "verify_certificate": _bool,
+    "default_reservation_period": _bool,
+    "loan_limit": int,
+    "hold_limit": int,
+    "max_retry_count": int,
+    "ebook_loan_duration": int,
+    "default_loan_duration": int,
+}
+
+
+def _coerce_types(settings: dict) -> None:
+    """Coerce the types, in-place"""
+    setting_type: Callable
+    for setting_name, setting_type in ALL_SETTING_TYPES.items():
+        if setting_name in settings:
+            settings[setting_name] = setting_type(settings[setting_name])
```

Comment:
Lets use pydantics type cohesion here instead of the `_bool` function. They provide [parse_obj_as](https://docs.pydantic.dev/1.10/usage/models/#parsing-data-into-a-specified-type), which should do a more robust conversion.

Something like: 
```python
for setting_name, setting_type in ALL_SETTING_TYPES.items():
    if setting_name in settings:
        settings[setting_name] = parse_obj_as(settings_type, settings[setting_name])
```